### PR TITLE
ABI: fix dynamic bytes location calculation

### PR DESCRIPTION
**Pull request by Jeiwan**
Currently, dynamic bytes location doesn't consider the size fields and elements of previous dynamic inputs. This PR fixes that.

### CODE REVIEW

**Comment by Jeiwan** [comment]
The failing tests are not caused by this PR.

**Comment by glitch003** [comment]
I merged this into my fork: https://github.com/deconet/ethereum.rb

File: spec/ethereum/encoder_spec.rb [hunk 1]
```diff
@@ -107,6 +107,12 @@
     it { expect(encoder.encode_arguments(function.inputs, ["dave", true])).to eq data }
   end
 
+  context "encode bytes function input following a dynamic input" do
```
Comment (kurotaky) [inline spec/ethereum/encoder_spec.rb]: I ran the test on my computer and succeeded.

**Comment by kurotaky** [comment]
@Jeiwan Thanks :)

**Pull request merged**